File: MPI/Esercizio01/helloworld.f90
!------------------------------------------!
!  Example: Hello                          !
!  All processes print an Hello World      !
!  with its rank                           !
!------------------------------------------!

      program main

      use mpi

      integer ierr,rank,procs

!--- initialize MP  -----------------------------------------
      call MPI_INIT(ierr)

!--- get my identifier in standard communicator -------------
      call MPI_COMM_RANK(MPI_COMM_WORLD,rank,ierr)

!--- get the number of processes in standard communicator ---
      call MPI_COMM_SIZE(MPI_COMM_WORLD,procs,ierr)

      print '(a,i2,a,i2)', 'Hello! I am process',rank  &
          ,' of ',procs


!--- finalize MPI  --------------------------.----------------
      call MPI_FINALIZE(ierr)

      end

File: MPI/Esercizio01/helloworld.c
/* Example: Hello
 *
 * All processes print an Hello World with its rank
*/  
#include <stdio.h>
#include "mpi.h"

int main(int argc, char **argv)
{
    int rank, size;
    
    /* initialize MPI */
    MPI_Init(&argc, &argv);

    /* get my identifier in standard communicator */
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);

    /* get the number of processes in standard communicator */
    MPI_Comm_size(MPI_COMM_WORLD, &size);

    printf("Hello! I am process %d of %d\n", rank, size);
     
    /* finalize MPI */
    MPI_Finalize();

    return 0;
}

